crates/release_plz_core/src/project.rs

```diff
@@ -171,21 +172,53 @@ impl Project {
             if package.description.is_none() {
                 missing_fields.push(format!("- `description` for package `{}`", package.name));
             }
-        }
 
-        if missing_fields.is_empty() {
-            Ok(())
-        } else {
-            let error_message = format!(
+            check_local_dependencies(package, &mut missing_versions);
+        }
+        let has_missing_fields = !missing_fields.is_empty();
+        let has_missing_version = !missing_versions.is_empty();
+        if !has_missing_fields && !has_missing_version {
+            return Ok(());
+        }
+        let mut error_message = String::new();
+        if has_missing_fields {
+            error_message.push_str(&format!(
                 "The following mandatory fields for crates.io are missing in Cargo.toml:
 {}
-See https://doc.rust-lang.org/cargo/reference/manifest.html
-
-Note: to disable this check, set the `--no-toml-check` flag.",
+See https://doc.rust-lang.org/cargo/reference/manifest.html\n",
                 missing_fields.join("\n")
-            );
-            anyhow::bail!(error_message);
+            ));
+        }
+        if has_missing_version {
+            error_message.push_str(&format!(
+                "The following packages have local dependencies missing a version specifier:
+{}",
+                missing_versions.join("\n")
+            ));
         }
+        error_message.push_str("\nNote: to disable this check, set the `--no-toml-check` flag.");
+        anyhow::bail!(error_message);
+    }
+}
+
+fn check_local_dependencies(package: &Package, missing_versions: &mut Vec<String>) {
+    //Check if version is specified for local dependencies (has a path entry)
+    let mut has_missing_version = false;
+    let mut missing_version = String::new();
+    for dependency in &package.dependencies {
+        if dependency.path.is_some() && dependency.req.comparators.is_empty() {
+            if !has_missing_version {
+                missing_version.push_str(&format!("- package `{}`:\n", package.name));
+            }
+            missing_version.push_str(&format!(
+                "\t• local dependency `{}` is missing a `version` entry\n",
+                dependency.name
+            ));
+            has_missing_version = true;
+        }
+    }
+    if has_missing_version {
+        missing_versions.push(missing_version);
```

Comment:
I would expect this function to return a vector of dependencies, i.e. `dependency.name` for the package. Than the actual error string is calculated outside of this function (e.g. in another function).
If you do it like this, you don't need the `missing_version` variable because in the other function you can add the string `- package `{}`:\n` if the vector is empty.

Reply:
Done.